load_data_json.js

```diff
@@ -0,0 +1,246 @@
+    body.appendChild(divElement)
+    
+    //CREATING NEW TABLE ELEMENT
+    createTable();
+
+    //FILL TABLE WITH USERS DATA
+    fillTable();
+
+    //Creating ADD BUTTONid="add"id="add"="add"id="add"d="add"
+    divElement=document.createElement("div")
+    divElement.align="center"
+    divElement.style.marginTop= "10px"
+    divElement.style.marginBottom= "10px"
+    divElement.innerHTML='<button class="button1" id="add" onclick="addData()">Add Data</button>';
+    body.appendChild(divElement)
+}
+
+//CREATES TABLE ELEMENT WITH HEADERS FROM tableHeaders ARRAY
+function createTable(){
+    
+    //CREATING DIV ELEMENT
+    let divElement=document.createElement("div")
+    divElement.id="table-div"
+    divElement.className="w3-container"
+
+    //CREATING TABLE ELEMENT
+    document.getElementsByTagName("body")[0].appendChild(divElement);
+    let myTable=document.createElement("table");
+    myTable.id="mytable"
+    myTable.className="w3-table-all w3-centered"
+    document.getElementById("table-div").appendChild(myTable);
+    let tableRow=myTable.insertRow();
+
+    //CREATING HEADERS OF TABLE
+    for(let i=0;i<tableHeaders.length;i++){
+        let tableCell=document.createElement("th");
+        tableRow.appendChild(tableCell);
+        tableCell.appendChild(document.createTextNode(tableHeaders[i]));
+    }
+}
+
+
+function fillTable(){
+    let mytable=document.getElementsByTagName("table")[0];
+
+    //INSERTING USER DATA IN TABLE
+    for(let i=0;i<usersData.length;i++){
+        let tableRow=mytable.insertRow()
+        tableRow.id="tr"+i;
+        for(let j in usersData[i]){
+            let tableCell=tableRow.insertCell()
+            tableCell.appendChild(document.createTextNode(usersData[i][j]));
+        }
+        // INSERTING EDIT AND DELETE BUTTTONS
+        let tableCell=tableRow.insertCell()
+        tableCell.innerHTML='<button class="w3-button w3-teal">Edit</button>'
+        tableCell.firstChild.onclick=edit
+        tableCell= tableRow.insertCell();
+        tableCell.innerHTML='<button class="w3-button w3-teal" onclick="del(this)">Delete</button>'
+        
+    }
+    
+}
+
+
+// TO EDIT A GIVEN ROW
+function edit(){
+    let currentTableRow=this.parentNode.parentNode
+   
+    let rowData=[] //TO STORE CURRENT ROW DATA IN CASE OF CANCEL
+    let currentColumn=currentTableRow.firstChild
+    let noOfCells=7;
+    while(noOfCells--){
+        rowData.push(currentColumn.textContent)
+        currentColumn=currentColumn.nextSibling
+    }
+
+    //MAKING ROWS EDITABLE
+    noOfCells=7
+    currentColumn=currentTableRow.firstChild
+    while(noOfCells--){
+        currentColumn.contentEditable=true
+        currentColumn=currentColumn.nextSibling
+    }
+
+    //CHANGING EDIT AND DELETE BUTTONS TO SAVE AND CANCEL
+    let delButtonCell=this.parentNode.nextSibling
+    this.textContent="Save"
+    this.onclick=function(){//ACTION ON CLICKING SAVE BUTTON
```

Comment:
Try to remove the listener when its not required anymore. Read about strong reference and weak reference and memory leaks in JS